index.js

```diff
@@ -161,9 +177,16 @@ tslintPlugin.report = function(reporter, options) {
     var throwErrors = function() {
         // Throw error
         if (options && options.emitError === true && errorFiles.length > 0) {
-            return this.emit('error', new PluginError('gulp-tslint', 'Failed to lint: ' + allFailures.map(function(failure) {
+            var start = options.reportLimit >= 0 ? options.reportLimit : 0;
+            var ignoreCount = options.reportLimit >= 0 ? allFailures.length - options.reportLimit : 0;
+            var failuresToOutput = allFailures.splice(start, ignoreCount);
+
+            var failureOutput = allFailures.map(function(failure) {
```

Comment:
This array contains all failures even if some were ignored, but then it's used in the return this.emit line. Did you mean to use the other variable perhaps?